plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java

```diff
@@ -212,22 +208,22 @@ private void returnFailure(BulkByScrollResponse response, String modelId, Action
         actionListener.onFailure(new OpenSearchStatusException(errorMessage, RestStatus.INTERNAL_SERVER_ERROR));
     }
 
-    private void deleteModel(String modelId, ActionListener<DeleteResponse> actionListener) {
-        DeleteRequest deleteRequest = new DeleteRequest(ML_MODEL_INDEX, modelId);
-        client.delete(deleteRequest, new ActionListener<DeleteResponse>() {
+    private void deleteModel(String modelId, FunctionName functionName, ActionListener<DeleteResponse> actionListener) {
+        DeleteRequest deleteRequest = new DeleteRequest(ML_MODEL_INDEX, modelId).setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+        client.delete(deleteRequest, new ActionListener<>() {
             @Override
             public void onResponse(DeleteResponse deleteResponse) {
-                deleteModelChunks(modelId, deleteResponse, actionListener);
+                if (FunctionName.REMOTE != functionName) {
+                    deleteModelChunks(modelId, deleteResponse, actionListener);
+                } else {
+                    actionListener.onResponse(deleteResponse);
+                }
                 deleteController(modelId);
             }
 
             @Override
             public void onFailure(Exception e) {
                 log.error("Failed to delete model meta data for model: " + modelId, e);
-                if (e instanceof ResourceNotFoundException) {
```

Comment:
I think we should not delete this part. This is for hand edge cases that model meta doc deleted but failed to delete model chunks.

Reply:
When catching exception: https://github.com/opensearch-project/ml-commons/blob/6bff20d5ea2dd16d40904d7eef1c22ffeda5b9ea/plugin/src/main/java/org/opensearch/ml/action/models/DeleteModelTransportAction.java#L228

Reply:
Can we return error response and give user the guidance how to clean up instead of we handle this for user, you can't ensure the clean up can  be succeed.

Reply:
> This is for hand edge cases that model meta doc deleted but failed to delete model chunks.

Not quite get you, what guidance we should give to user if model meta deleted ?

Reply:
I think the correct direction is: When deleting a model, we should delete model meta and chunks and controller, only when all three are deleted we return success, otherwise we return failure, and tell user `model chunks or controller not cleaned up, please try again.`, and user can try delete model chunks and controller until succeed.

Reply:
Make sense. I think the old logic is the same with your idea, we delete model meta first, then delete chunks, if anyone failed, we throw exception. Took a look at code, the model controller part will not throw exception when delete model. You can fix this.

Reply:
DeleteModelTransportAction.java
` deleteController(modelId);` doesn't have `actionListener` , that looks not reasonable 
```
private void deleteModel(String modelId, ActionListener<DeleteResponse> actionListener) {
        DeleteRequest deleteRequest = new DeleteRequest(ML_MODEL_INDEX, modelId);
        client.delete(deleteRequest, new ActionListener<DeleteResponse>() {
            @Override
            public void onResponse(DeleteResponse deleteResponse) {
                deleteModelChunks(modelId, deleteResponse, actionListener);
                deleteController(modelId);
            }

            @Override
            public void onFailure(Exception e) {
                log.error("Failed to delete model meta data for model: " + modelId, e);
                if (e instanceof ResourceNotFoundException) {
                    deleteModelChunks(modelId, null, actionListener);
                    deleteController(modelId);
                }
                actionListener.onFailure(e);
            }
        });
    }
```